xtable-api/src/main/java/org/apache/xtable/model/storage/InternalDataFile.java

```diff
@@ -33,17 +38,18 @@
  *
  * @since 0.1
  */
-@Builder(toBuilder = true)
-@Value
-public class InternalDataFile {
-  // physical path of the file (absolute with scheme)
-  @NonNull String physicalPath;
+@SuperBuilder(toBuilder = true)
+@FieldDefaults(makeFinal = true, level = lombok.AccessLevel.PRIVATE)
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@Accessors(fluent = true)
+@Getter
+public class InternalDataFile extends InternalBaseFile {
   // file format
   @Builder.Default @NonNull FileFormat fileFormat = FileFormat.APACHE_PARQUET;
   // partition ranges for the data file
   @Builder.Default @NonNull List<PartitionValue> partitionValues = Collections.emptyList();
 
-  long fileSizeBytes;
   long recordCount;
```

Comment:
Should the count be moved to the base as well? there is `countRecordsDeleted` mentioned in the RFC

Reply:
The record count doesn't apply to stat files. However, as mentioned above, XTable currently supports delete and data files, both of which have record counts. So, this seems reasonable.